framework/core/src/User/TokensClearer.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User;
+
+use Flarum\User\Event\EmailChanged;
+use Flarum\User\Event\PasswordChanged;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class TokensClearer
+{
+    public function subscribe(Dispatcher $events): void
+    {
+        $events->listen(PasswordChanged::class, [$this, 'clearPasswordTokens']);
```

Comment:
```suggestion
        $events->listen([PasswordChanged::class, EmailChanged::class], [$this, 'clearPasswordTokens']);
```

This helps keep method declaration in one spot. Line 22 needs to be removed though.

Not sure which format you prefer, but using an array of events in `listen()` is supported.